### PR TITLE
Backport ASA-2024-001

### PR DESCRIPTION
**Pull request by sergio-mena**
Manual backport of fix and UTs of advisory ASA-2024-001

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: types/params.go [hunk 1]
```diff
@@ -205,27 +205,58 @@ func (params ConsensusParams) ValidateBasic() error {
...
+	if updated == nil || updated.Abci == nil {
 		return nil
 	}
-	if params.ABCI.VoteExtensionsEnableHeight == updated.Abci.VoteExtensionsEnableHeight {
+	// 2
+	if updated.Abci.VoteExtensionsEnableHeight < 0 {
+		return errors.New("VoteExtensionsEnableHeight must be positive")
+	}
+	// 3
+	if params.ABCI.VoteExtensionsEnableHeight <= 0 && updated.Abci.VoteExtensionsEnableHeight == 0 {
```
Comment (mzabaluev) [inline types/params.go:229]: Side note: why do we use all-caps in the human-written field name (that's not what you should do to acronyms per CamelCase), when protobuf has it differently?

**Review by mzabaluev — APPROVED** [review]
Looks good.

File: types/params.go (hunk 1 above)
Comment (sergio-mena) [reply types/params.go:229]: Good point. But I think it's too late to change that now for backwards compatibility

Comment (sergio-mena) [reply types/params.go:229]: Also ABCI is an acronym, so not sure how CameCase usually deals with that

Comment (mzabaluev) [reply types/params.go:229]: > Also ABCI is an acronym, so not sure how CameCase usually deals with that

At least in Rust, you make it a "word" e.g. `TcpStream`.

Comment (mzabaluev) [reply types/params.go:229]: > it's too late to change that now for backwards compatibility

Or the last opportunity to clean up the style prior to the 1.0 release? :wink:

Comment (greg-szabo) [reply types/params.go:229]: Yeah, the `params.ABCI` vs `updated.Abci` got me in the first half. It's not something we can fix here but I can open an issue with your side note, if we want to fix it.

**Review by greg-szabo — APPROVED** [review]

**Pull request merged**